examples/advanced/example_model_optimization.py

```diff
@@ -339,19 +324,49 @@ def suppress_output():
 
         with suppress_output():
             state_dict = hub.load_state_dict_from_url(url)
+        return state_dict
+
 
+use_example_transformer_weights = True
+if use_example_transformer_weights:
+    state_dict = download_example_transformer_weights()
     transformer.load_state_dict(state_dict)
-else:
-    dataset = None
-    transformer = train(transformer, dataset)
-
-    state_dict = OrderedDict(
-        [
-            (name, parameter.detach().cpu())
-            for name, parameter in transformer.state_dict().items()
-        ]
-    )
-    torch.save(state_dict, checkpoint)
+
+########################################################################################
+#
+#
+# .. note::
+#
+#   The weights of the provided transformer were trained with the
+#   `2014 training images <http://images.cocodataset.org/zips/train2014.zip>`_ of the
+#   `COCO dataset <https://cocodataset.org/>`_. The training was performed for
+#   ``num_epochs=2`` and ``batch_size=4``. Each image was center-cropped to
+#   ``256 x 256`` pixels.
+#
+# .. note::
+#
+#   If you want to perform the training yourself, set ``root`` to a location
+#   of a folder of images.
+
+
+root = None
+checkpoint = "example_transformer.pth"
+if not use_example_transformer_weights:
+    if root:
+        transformer = train(
+            transformer=transformer, root=root, batch_size=4, epochs=2, image_size=256
```

Comment:
Nit: put a comma there so the auto-formatter (`tox -e lint-style`) splits this into multiple lines and enhances readability

```suggestion
            transformer=transformer, root=root, batch_size=4, epochs=2, image_size=256,
```